src/components/MenuDelivery/MenuDelivery.jsx

```diff
@@ -48,15 +53,42 @@ const StyledMenu = styled((props) => (
   },
 }));
 
-const MenuDelivery = () => {
-  const [anchorEl, setAnchorEl] = React.useState(null);
+const MenuDelivery = ({ orderId, deliveryHistory, setDeliveryHistory }) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+
+  const [openModalDelete, setOpenModalDelete] = useState(false);
+
+  const openDelete = () => setOpenModalDelete(true);
+
+  const closeDelete = () => {
+    setOpenModalDelete(false);
+    handleClose();
+  };
+
   const open = Boolean(anchorEl);
+
   const handleClick = (event) => {
     setAnchorEl(event.currentTarget);
   };
+
   const handleClose = () => {
     setAnchorEl(null);
   };
+  const { update } = DeliveryStorageService;
+
+  const deleteRow = (id) => {
+    const rowToDelete = deliveryHistory.filter((el) => el.orderId === id);
+    const filterRows = deliveryHistory.filter(
+      (el) => el.orderId !== rowToDelete[0].orderId
```

Comment:
With this change the filter above is not needed
```suggestion
      (el) => el.orderId !== id
```

Reply:
Oh very true.